trunk/source/IcfgBuilder/src/de/uni_freiburg/informatik/ultimate/plugins/generator/icfgbuilder/cfg/CfgBuilder.java

```diff
@@ -1672,9 +1692,13 @@ private void mergeLocNodes(final BoogieIcfgLocation oldLocNode, final BoogieIcfg
 				mIcfg.getProcedureEntryNodes().put(mCurrentProcedureName, newLocNode);
 			}
 			if (mIcfg.getLoopLocations().remove(oldLocNode)) {
-				// if the old location was a loop location, the new one is also
+				// if the old location was a loop location, the new one is also a loop location
 				mIcfg.getLoopLocations().add(newLocNode);
```

Comment:
Now there are two sets of locations: Locations of interest and loop locations.
Therefore I wonder: Do we still need the loop locations? And if yes, should they be also contained in the locations of interest?

Reply:
* Yes, we still need loop locations. E.g., for debugging, termination analysis, providing messages where the nodes stems from. It might be option to get rid of loop locations. But this should be well thought out in advance and needs modifications in other places.
* Loop heads should not be contained in the set of LOIs. This is a rather random decision, motivated by the fact that I do not want to introduce too many changes.